pkg/network/event_common.go

```diff
@@ -178,8 +178,10 @@ func (c ConnectionStats) ByteKey(buf [ConnectionByteKeyMaxLen]byte) ([]byte, err
 	buf[n] = uint8(c.Family)<<4 | uint8(c.Type)
 	n++
 
-	n += copy(buf[n:], c.Source.Bytes()) // 4 or 16 bytes
-	n += copy(buf[n:], c.Dest.Bytes())   // 4 or 16 bytes
+	i, _ := c.Source.Write(buf[n:]) // 4 or 16 bytes
+	n += i
+	i, _ = c.Dest.Write(buf[n:]) // 4 or 16 bytes
+	n += i
```

Comment:
I was a bit surprised that Bytes() method was generating one allocation per call.
This is how the implementation looks like:

```go
type v4Address [4]byte

func (a v4Address) Bytes() []byte {
	return a[:]
}
```
Could someone explain why this is allocating?

Reply:
Its allocating a slice header.   Likely the escape analysis isn't smart enough to determine that the slice wasn't escaping.